ledger/participant-integration-api/src/main/scala/platform/indexer/ExecuteUpdate.scala

```diff
@@ -429,11 +428,14 @@ class PipelinedExecuteUpdate(
   private[indexer] val flow: ExecuteUpdateFlow =
     Flow[OffsetUpdate]
       .mapAsync(updatePreparationParallelism)(prepareUpdate)
-      .buffer(16, OverflowStrategy.backpressure)
+      .async
       .map(PipelinedUpdateWithTimer(_, metrics.daml.index.db.storeTransaction.time()))
       .mapAsync(1)(insertTransactionState)
+      .async
```

Comment:
Do you know what's the improvement of adding these async barriers? I considered adding them a while ago, but decided against since I didn't believe they could bring anything when we have a buffer before the insertion stages.

Reply:
Without the async barriers, the three mapAsync stages are being fused together, causing some additional overhead that is not reflected in the individual metrics.